_posts/2020-05-25-flink-on-zeppelin-part2.md

```diff
@@ -0,0 +1,107 @@
+---
+layout: post
+title:  "Flink on Zeppelin Notebooks for Interactive Data Analysis - Part 2"
+date:   2020-05-25T08:00:00.000Z
+categories: ecosystem
+authors:
+- zjffdu:
+  name: "Jeff Zhang"
+  twitter: "zjffdu"
+---
+
+In the last post, I introduce the basics of Flink on Zeppelin and how to do Streaming ETL. This is part-2 where I would talk about how to 
+do streaming data visualization via Flink on Zeppelin and how to use flink UDF in Zeppelin. 
+
+# Streaming Data Visualization
+
+In Zeppelin, you can build a realtime streaming dashboard without writing any line of javascript/html/css code.
+Overall Zeppelin supports 3 kinds of streaming data analytics:
+* Single
+* Update
+* Append
+
+### Single Mode
+Single mode is for the case when the result of sql statement is always one row, such as the following example. 
+The output format is HTML, and you can specify paragraph local property template for the final output content template. 
+And you can use {i} as placeholder for the ith column of result.
```

Comment:
I am not sure what "ith" maybe you want to ad {i}th if you want to say 9th or 8th? 
It should also be "column of **the** result"